### PR TITLE
Add Windows support

### PR DESCRIPTION
**Pull request by mondeja**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -335,4 +390,24 @@ mod tests {
...
+    #[test]
+    fn test_parse_hostfile() {
+        let maybe_hostfile = parse_hostfile();
+        assert!(maybe_hostfile.is_ok());
+        let hostfile = maybe_hostfile.unwrap();
+        assert!(!hostfile.is_empty());
+
+        #[cfg(not(windows))]
+        {
+            // Unix systems *tipically* include localhost in /etc/hosts.
```
Comment (aneeshdurg) [inline src/lib.rs]:
```suggestion
            // Unix systems *typically* include localhost in /etc/hosts.
```

File: src/lib.rs [hunk 2]
```diff
@@ -335,4 +390,24 @@ mod tests {
         let cloned = host_entry.clone();
         assert_eq!(host_entry, cloned)
     }
+
+    #[test]
+    fn test_parse_hostfile() {
```
Comment (aneeshdurg) [inline src/lib.rs:415]: This test seems to depend on Github Action's hostile specifically right? Ideally I'd like for these tests to work on any environment, but I can see the value of parsing the file for real (and validating that we choose the right path on windows vs Linux). 

Can we add an explicit comment at the top of this test saying that this test assumes that it's being run on github actions?

**Review by aneeshdurg — COMMENTED** [review]
Thank you for these changes! Left some minor comments - should be good to merge after they're resolved.

File: src/lib.rs (hunk 2 above)
Comment (mondeja) [reply src/lib.rs:415]: Okay, now that part is only executed on GHA.

Note that still assumes that there is one entry but the default hostfile in Windows is empty. Do you wan't to remove that assumption too? I can separate the logic of discovering the file path, just test their existence and make this test completely GHA-dependant.

Comment (mondeja) [reply src/lib.rs:395]: Let's do that instead...

Comment (mondeja) [reply src/lib.rs:415]: Should be fine now.

**Review by aneeshdurg — APPROVED** [review]
LGTM!

**Comment by aneeshdurg** [comment]
Looks like the windows tests are failing - seems like the return type of `get_hostfile_path` should be `Result<Path, String>` - will merge when tests pass

**Comment by mondeja** [comment]
It's fixed 👍🏼

**Pull request merged**